src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -156,13 +156,45 @@ trait RegularSync {
     }
   }
 
+  /**
+    * Inserts and executes the transactions of the blocks, up to the point to which one of them fails (or we run out of blocks).
+    * If the execution of any block were to fail, newBlocks only contains the NewBlock msgs for all the blocks executed before it.
+    *
+    * @param blocks to execute
+    * @param blockParentTd, td of the parent of the blocks.head block
+    * @param newBlocks which, after adding the corresponding NewBlock msg for blocks, will be broadcasted
+    * @return list of NewBlocks to broadcast (one per block successfully executed) and an error if one happened during execution
+    */
+  @tailrec
+  private def insertBlocks(blocks: Seq[Block], blockParentTd: BigInt,
```

Comment:
I haven't formally checked that but it does seem to take longer to run regular sync with tx execution enabled than not so it should be worth taking a look at it. Shall we create a task for checking that? As this PR already contains changes to many files I would prefer to take a look at it in another PR